src/main/scala/com/example/todoapp/endpoints/SpaEndpoints.scala

```diff
@@ -0,0 +1,24 @@
+package com.example.todoapp.endpoints
+
+import cats.effect.IO
+import sttp.tapir.*
+import scala.io.Source
+import sttp.tapir.server.ServerEndpoint
+
+object SpaEndpoints {
+  private object SpaServerLogics {
+    def angularAppLogic: List[String] => IO[Either[Unit, String]] = _ =>
+      IO {
+        val html = Source.fromFile("public/ngx/browser/index.html").mkString
+        Right(html)
+      }
+  }
+
+  private def angularAppEndpoint: PublicEndpoint[List[String], Unit, String, Any] = {
+    endpoint.get.in(paths).out(htmlBodyUtf8)
+  }
+
+  val endpoints: List[ServerEndpoint[Any, IO]] = List(
+    angularAppEndpoint.serverLogic(SpaServerLogics.angularAppLogic)
+  )
```

Comment:
`sttp.tapir.files.staticFilesGetServerEndpoint` 使った方がシンプルになるかと。

```suggestion
  val endpoints: List[ServerEndpoint[Any, IO]] = List(
    staticFilesGetServerEndpoint("")("public/ngx/browser/index.html")
  )
```

Reply:
私もそう思い `staticFilesGetServerEntpoint` を用いて実装してみたのですが、「4.6.2 Angular アプリケーションの配信」で示されている

> http4s のサーバーにおいて、/ping /api/* 以外のパスに遷移すると /public/ngx/browser/index.html を返却し、静的コンテンツを配信できるようにしてください。

という要件が満たせなくなってしまいます。(`http://localhost:9000/` 以外でこのページが開けなくなります)
説明によると、今回は http4s から tapir へ置き換えるという内容だったので、この要件も満たされ続けるべきだと解釈しましたが、`http://localhost:9000/` にアクセスした時に表示されれば十分でしょうか？

Reply:
@rito528 こちら、元の要件がちょっとよくなかったかなと思うのでこの修正で問題なしとします 🙇